tests/FINDOLOGIC/Api/Tests/RequestBuilders/Json/SuggestionRequestBuilderTest.php

```diff
@@ -0,0 +1,192 @@
+
+        $searchRequestBuilder->setQuery($expectedQuery);
+        $params = $searchRequestBuilder->getParams();
+        $this->assertArrayHasKey($expectedParameter, $params);
+        $this->assertEquals($expectedQuery, $params[$expectedParameter]);
+    }
+
+    /**
+     * @dataProvider invalidQueryProvider
+     * @param mixed $invalidQuery
+     */
+    public function testSetQueryWillThrowAnExceptionWhenSubmittingInvalidQueries($invalidQuery)
+    {
+        $this->expectException(InvalidParamException::class);
+        $this->expectExceptionMessage('Parameter query is not valid.');
+
+        $searchRequestBuilder = new SuggestRequestBuilder();
+        $this->setRequiredParamsForSuggestionRequestBuilder($searchRequestBuilder);
+
+        $searchRequestBuilder->setQuery($invalidQuery);
+    }
+
+    /**
+     * @dataProvider autocompleteBlocksProvider
+     * @param string $autocompleteBlock
+     */
+    public function testAddAutocompleteBlocksWillSetItInAValidFormat($autocompleteBlock)
+    {
+        $expectedParameter = 'autocompleteblocks';
+
+        $searchRequestBuilder = new SuggestRequestBuilder();
+        $this->setRequiredParamsForSuggestionRequestBuilder($searchRequestBuilder);
+
+        $searchRequestBuilder->addAutocompleteBlocks($autocompleteBlock);
+        $params = $searchRequestBuilder->getParams();
+        $this->assertArrayHasKey($expectedParameter, $params);
+        $this->assertEquals($autocompleteBlock, $params[$expectedParameter]);
+    }
+
+    /**
+     * @dataProvider invalidAutocompleteBlocksProvider
+     * @param mixed $invalidAutocompleteblock
+     */
+    public function testInvalidShopurlWillThrowAnException($invalidAutocompleteblock)
+    {
+        $this->expectException(InvalidParamException::class);
+        $this->expectExceptionMessage('Parameter autocompleteblocks is not valid.');
+
+        $searchRequestBuilder = new SuggestRequestBuilder();
+        $this->setRequiredParamsForSuggestionRequestBuilder($searchRequestBuilder);
+
+        $searchRequestBuilder->addAutocompleteBlocks($invalidAutocompleteblock);
+    }
+
+    /**
+     * @dataProvider usergroupProvider
+     * @param string $usergroup
+     */
+    public function testSetUsergroupHashWillSetItInAValidFormat($usergroup)
+    {
+        $expectedParameter = 'usergrouphash';
+
+        $searchRequestBuilder = new SuggestRequestBuilder();
+        $this->setRequiredParamsForSuggestionRequestBuilder($searchRequestBuilder);
+
+        $searchRequestBuilder->setUsergrouphash($usergroup);
+        $params = $searchRequestBuilder->getParams();
+        $this->assertArrayHasKey($expectedParameter, $params);
+        $this->assertEquals($usergroup, $params[$expectedParameter]);
+    }
+
+    /**
+     * @dataProvider invalidUsergroupProvider
+     * @param mixed $invalidUsergroup
+     */
+    public function testInvalidUsergroupHashlWillThrowAnException($invalidUsergroup)
```

Comment:
What is `…HashlWill…`?

Reply:
That `l` slipped in therelllllllllllllllllllllllll. x)